File: ITapply/ITapply.Models/Requests/CandidateInsertRequest.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using static ITapply.Models.Responses.EnumResponse;

namespace ITapply.Models.Requests
{
    public class CandidateInsertRequest
    {
        [Required]
        public int UserId { get; set; }

        [Required]
        [StringLength(100)]
        public string FirstName { get; set; }

        [Required]
        [StringLength(100)]
        public string LastName { get; set; }

        [StringLength(20)]
        public string PhoneNumber { get; set; }

        [StringLength(100)]
        public string Title { get; set; }

        [StringLength(2000)]
        public string Bio { get; set; }

        public int? LocationId { get; set; }

        [Range(0, 100)]
        public int ExperienceYears { get; set; }

        public ExperienceLevel ExperienceLevel { get; set; }
    }
} 

File: ITapply/ITapply.UI/itapply_mobile/lib/models/requests/candidate_insert_request.dart
import 'package:itapply_desktop/models/enums.dart';
import 'package:json_annotation/json_annotation.dart';

part 'candidate_insert_request.g.dart';

@JsonSerializable()
class CandidateInsertRequest {
  final int userId;
  final String firstName;
  final String lastName;
  final String? phoneNumber;
  final String? title;
  final String? bio;
  final int? locationId;
  final int? experienceYears;
  @JsonKey(toJson: _experienceLevelToJson)
  final ExperienceLevel? experienceLevel;

  CandidateInsertRequest({
    required this.userId,
    required this.firstName,
    required this.lastName,
    this.phoneNumber,
    this.title,
    this.bio,
    this.locationId,
    this.experienceYears,
    this.experienceLevel,
  });

  factory CandidateInsertRequest.fromJson(Map<String, dynamic> json) => _$CandidateInsertRequestFromJson(json);
  Map<String, dynamic> toJson() => _$CandidateInsertRequestToJson(this);
}

int? _experienceLevelToJson(ExperienceLevel? level) => level?.index;